packages/node-resolve/test/browser.js

```diff
@@ -199,3 +199,33 @@ test('pkg.browser with mapping to prevent bundle by specifying a value of false'
 
   t.is(module.exports, 'ok');
 });
+
+test('pkg.browser can override the export map result', async (t) => {
+  const bundle = await rollup({
+    input: 'browser-override-exports.js',
+    plugins: [nodeResolve({ browser: true }), commonjs()]
+  });
+  const { module } = await testBundle(t, bundle);
+
+  t.is(module.exports, 'browser');
+});
+
+test('exports.browser takes precedence over export map result, when browser:true', async (t) => {
+  const bundle = await rollup({
+    input: 'browser-exports-browser.js',
+    plugins: [nodeResolve({ browser: true }), commonjs()]
+  });
+  const { module } = await testBundle(t, bundle);
+
+  t.is(module.exports, 'browser');
+});
+
+test('exports.browser does not take precedence over export map result, when browser:false', async (t) => {
+  const bundle = await rollup({
+    input: 'browser-exports-browser.js',
+    plugins: [nodeResolve(), commonjs()]
+  });
+  const { module } = await testBundle(t, bundle);
+
+  t.is(module.exports, 'require');
```

Comment:
Somewhat suprised to see this is returning `require` by default, I wasn't aware this was the case, it might be a separate issue if not introduced by this PR (in which case the previous comment should also be `require` I suppose). We should surely be defaulting to ESM for top-level imports.

Reply:
Maybe it's because I included the `commonjs` plugin as well? That is copy-pasta by the way, I'll just try removing it :)

Reply:
I'm not sure, perhaps @LarsDenBakker can clarify what the expectation is for this case.

Reply:
FWIW, removing the commonjs plugin made no difference.

Reply:
That makes sense, it's about the top-level resolver conditions, not the transforms.